.expeditor/scripts/build-infra-deb.sh

```diff
@@ -0,0 +1,187 @@
+
+    echo "Unpacking migration tool..."
+    tar -xf "$TEMP_DIR/migration-tools.tar.gz" -C "$PACKAGE_DIR$CHEF_BIN_DIR/" || { echo "Error: Failed to unpack migration tool"; exit 1; }
+
+    echo "Copying Chef Infra tarball..."
+    cp "$TEMP_DIR/chef-infra-client.tar.gz" "$PACKAGE_DIR$CHEF_BUNDLE_DIR/" || { echo "Error: Failed to copy Chef Infra tarball"; exit 1; }
+
+    prepare_control_file
+    prepare_preinstall_script
+    prepare_postinstall_script
+    echo "Package structure prepared successfully."
+}
+
+prepare_control_file() {
+    cat <<EOL > "$PACKAGE_DIR/DEBIAN/control"
+Package: $DEB_PKG_NAME
+Version: $VERSION
+Architecture: $ARCH
+Maintainer: The Chef Maintainers <maintainers@chef.io>
+Description: Chef Infra Client
+EOL
+    if [ $? -ne 0 ]; then
+        echo "Error: Failed to create control file"; exit 1
+    fi
+    echo "Control file prepared successfully."
+}
+
+prepare_preinstall_script() {
+    cat <<EOL > "$PACKAGE_DIR/DEBIAN/preinst"
+#!/bin/bash
+
+FRESH_INSTALL_FLAG_FILE="/tmp/chef_fresh_install"
+
+if [ -d "/opt/chef" ]; then
+    rm -f "\$FRESH_INSTALL_FLAG_FILE"
+else
+    echo "fresh_install" > "\$FRESH_INSTALL_FLAG_FILE"
+fi
+
+if [[ "\$1" == "--help" ]]; then
+    echo -e "\nChef Infra Client Installation Help"
+    echo "Usage: sudo LICENSE_KEY=\"<license-key>\" dpkg -i <deb-file>"
+    echo "Example:"
+    echo "  sudo LICENSE_KEY=\"your-license-key\" dpkg -i chef-chef-infra-client-<version>-<timestamp>.deb"
+    echo "Environment Variables:"
+    echo "  LICENSE_KEY: The license key required for the Chef Infra Client installation."
+    exit 0
+fi
+
+if [ -z "\${LICENSE_KEY:-}" ]; then
+    LICENSE_KEY=$(env | grep -m 1 '^LICENSE_KEY=' | cut -d '=' -f 2 | xargs)
+    
+    if [ -z "\$LICENSE_KEY" ]; then
+        echo -e "\nError: LICENSE_KEY environment variable is required."
+        echo "Usage: sudo LICENSE_KEY=\"<license-key>\" dpkg -i <deb-file>"
+        exit 1
+    fi
+fi
+
+EOL
+    chmod +x "$PACKAGE_DIR/DEBIAN/preinst" || { echo "Error: Failed to make preinst script executable"; exit 1; }
+    echo "Pre-install script prepared successfully."
+}
+
+
+
+prepare_postinstall_script() {
+    cat <<EOL > "$PACKAGE_DIR/DEBIAN/postinst"
+#!/bin/bash
+CHEF_BIN_DIR="/opt/chef/bin"
+CHEF_BUNDLE_DIR="/opt/chef/bundle"
+FRESH_INSTALL_FLAG_FILE="/tmp/chef_fresh_install"
+
+FRESH_INSTALL_FLAG=""
+LICENSE_SERVER=\${CHEF_INFRA_LICENSE_SERVER:-}
+LICENSE_KEY=\${CHEF_INFRA_LICENSE_KEY:-}
+
+if [ -f "\$FRESH_INSTALL_FLAG_FILE" ]; then
+    FRESH_INSTALL_FLAG="--fresh_install"
+    echo "Postinstall: Detected fresh installation."
+    rm -f "\$FRESH_INSTALL_FLAG_FILE"
+else
+    echo "Postinstall: Detected upgrade installation."
+fi
+
+if [ -f "\$CHEF_BIN_DIR/chef-migrate" ]; then
+    echo "Running post-install tasks..."
+    MIGRATE_CMD="\$CHEF_BIN_DIR/chef-migrate apply airgap \$FRESH_INSTALL_FLAG \$CHEF_BUNDLE_DIR/chef-infra-client.tar.gz"
+
+    if [ -n "\$LICENSE_KEY" ] && [ -n "\$LICENSE_SERVER" ]; then
```

Comment:
Split it into two conditional statements: one to set the server and another to set the key.